compiler/z/codegen/S390GenerateInstructions.cpp

```diff
@@ -1006,11 +1006,21 @@ TR::Instruction *
 generateRSInstruction(TR::CodeGenerator * cg, TR::InstOpCode::Mnemonic op, TR::Node * n, TR::Register * treg, TR::Register * sreg, uint32_t imm,
                       TR::Instruction * preced)
    {
-   if (preced)
+   auto instructionFormat = TR::InstOpCode(op).getInstructionFormat();
+   TR::Instruction *result = NULL;
+   if (instructionFormat == RSa_FORMAT)
```

Comment:
I think we should also check for `RSb_FORMAT` here to be conservative -- in case someone wants to use ICM for example against one of the low-address zOS control blocks for example.

Reply:
Do we have RSb_Instruction with immediate source operand or possible to have one? Looking at the other RSb_Format, all seems to be using memory reference as second operand. If we should only use RSb_FORMAT with memory reference, than least I should do is to put a FATAL ASSERT to  make sure this function is only called for RSa_FORMAT or RSYa_FORMAT instruction.

Reply:
Thanks @joransiu  for the example. I do see that use case now.  Made changes in https://github.com/eclipse/omr/compare/92c477763e1045da785afc6cc535d13e8eb64272..ddc0bcf5dc7276dd5b7f8cb32954cfc001673191

Also included `RSYb_FORMAT` as well to generate `RSYInstruction` with immediate source operand instruction.